x/stakeibc/keeper/update_redemption_rates_test.go

```diff
@@ -0,0 +1,272 @@
+		stAtomBalance: sdk.NewInt64Coin(stAtom, int64(ST_SUPPLY)),
+	}
+	s.FundAccount(user.acc, user.stAtomBalance)
+
+	// set the staked balance on the host zone
+	hostZone := stakeibc.HostZone{
+		ChainId:        "GAIA",
+		HostDenom:      "uatom",
+		StakedBal:      STAKED_BAL,
+		RedemptionRate: REDEMPTION_RATE_T0,
+	}
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx, hostZone)
+
+	return UpdateRedemptionRatesTestCase{
+		host_zone:          hostZone,
+		redemption_rate_t0: REDEMPTION_RATE_T0,
+		staked_bal:         STAKED_BAL,
+		undelegated_bal:    UNDELEGATED_BAL,
+		just_deposited_bal: JUST_DEPOSITED_BAL,
+		st_supply:          ST_SUPPLY,
+	}
+}
+
+func (s *KeeperTestSuite) TestUpdateRedemptionRatesSuccessful() {
+
+	STAKED_BAL := uint64(5)
+	UNDELEGATED_BAL := uint64(3)
+	JUST_DEPOSITED_BAL := uint64(3)
+	ST_SUPPLY := uint64(10)
+
+	REDEMPTION_RATE_T0 := sdk.NewDec(1)
+	tc := s.SetupUpdateRedemptionRates(STAKED_BAL, UNDELEGATED_BAL, JUST_DEPOSITED_BAL, ST_SUPPLY, REDEMPTION_RATE_T0)
+
+	// sanity check on inputs (check redemptionRate at genesis is 1)
+	s.Require().Equal(tc.redemption_rate_t0, sdk.NewDec(1))
+
+	records := s.App.RecordsKeeper.GetAllDepositRecord(s.Ctx)
+	s.App.StakeibcKeeper.UpdateRedemptionRates(s.Ctx, records)
+
+	hz, found := s.App.StakeibcKeeper.GetHostZone(s.Ctx, tc.host_zone.ChainId)
+	s.Require().True(found)
+	rrNew := hz.RedemptionRate
+
+	s.Require().Equal(rrNew, sdk.NewDec(int64(STAKED_BAL)+int64(UNDELEGATED_BAL)+int64(JUST_DEPOSITED_BAL)).Quo(sdk.NewDec(int64(ST_SUPPLY))))
+	s.Require().Equal(rrNew, sdk.NewDec(11).Quo(sdk.NewDec(10)))
+}
+
+func (s *KeeperTestSuite) TestUpdateRedemptionRatesRandomized() {
+	// run N tests, each with random inputs
+
+	// set rand seed for consistency
+	genRandUintBelowMax := func(MAX int) uint64 {
+		MIN := int(0)
+		n := 0 + rand.Intn(MAX-MIN+1)
+		return uint64(n)
+	}
+
+	MAX := 1_000_000_000
+	STAKED_BAL := genRandUintBelowMax(MAX)
+	UNDELEGATED_BAL := genRandUintBelowMax(MAX)
+	JUST_DEPOSITED_BAL := genRandUintBelowMax(MAX)
+
+	ST_SUPPLY := genRandUintBelowMax(MAX)
+
+	// s.Require().ElementsMatch([]int{0, 0, 0, 0}, []int{int(STAKED_BAL), int(UNDELEGATED_BAL), int(JUST_DEPOSITED_BAL), int(ST_SUPPLY)}) //
+	REDEMPTION_RATE_T0 := sdk.NewDec(1)
+	tc := s.SetupUpdateRedemptionRates(STAKED_BAL, UNDELEGATED_BAL, JUST_DEPOSITED_BAL, ST_SUPPLY, REDEMPTION_RATE_T0)
+
+	// sanity check on inputs (check redemptionRate at genesis is 1)
+	s.Require().Equal(tc.redemption_rate_t0, sdk.NewDec(1))
+
+	records := s.App.RecordsKeeper.GetAllDepositRecord(s.Ctx)
+	s.App.StakeibcKeeper.UpdateRedemptionRates(s.Ctx, records)
+
+	hz, found := s.App.StakeibcKeeper.GetHostZone(s.Ctx, tc.host_zone.ChainId)
+	s.Require().True(found)
+	rrNew := hz.RedemptionRate
+
+	numerator := int64(STAKED_BAL) + int64(UNDELEGATED_BAL) + int64(JUST_DEPOSITED_BAL)
+	denominator := int64(ST_SUPPLY)
+	expectedNewRate := sdk.NewDec(numerator).Quo(sdk.NewDec(denominator))
+	s.Require().Equal(rrNew, expectedNewRate)
```

Comment:
since these inputs are randomized, I'd log them here by adding the description argument to `.Equal`, otherwise if this fails, you wont know the values that caused it to fail

Reply:
good idea, done